platform/macos/macos.xcodeproj/project.pbxproj

```diff
@@ -271,6 +274,10 @@
 /* End PBXCopyFilesBuildPhase section */
 
 /* Begin PBXFileReference section */
+		07A019EA1ED662D800ACD43E /* MGLImageSource_Private.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; path = MGLImageSource_Private.h; sourceTree = "<group>"; };
```

Comment:
That’s the correct (only) way to do it; however, the macOS project is currently separate from the iOS project, even though they share many files. So you need to also open up `make xproj` and delete the (now broken) reference to the file from that project as well. #5942 would merge the two projects, avoiding this complication in the future.